stat_maker.py

```diff
@@ -0,0 +1,215 @@
+                    except AttributeError:
+                        continue
+                else:
+                    for key, value in [('queue_name', queue_name),
+                                       ('run', run),
+                                       ('queued', queued),
+                                       ('lm', lm),
+                                       ('state', state)]:
+                        temp_dict[key] = value
+                    self.l.append(temp_dict)
+            self.l.append({'Total running': total_running_jobs, 'Total queued': total_queued_jobs})
+        self.dump_all(out_file, self.statq_mapping[write_method])
+
+    @staticmethod
+    def process_line(re_search, line, re_match_positions):
+        qstat_values = dict()
+        m = re.search(re_search, line.strip())
+        try:
+            job_id, user, job_state, queue = [m.group(x) for x in re_match_positions]
+        except AttributeError:
+            print line.strip()
+            sys.exit(0)
+        job_id = job_id.split('.')[0]
+        for key, value in [('JobId', job_id), ('UnixAccount', user), ('S', job_state), ('Queue', queue)]:
+            qstat_values[key] = value
+        return qstat_values
+
+
+class OarStatMaker(QStatMaker):
+    def __init__(self):
+        StatMaker.__init__(self)
+        self.user_q_search = r'^(?P<job_id>[0-9]+)\s+' \
+                             r'(?P<name>[0-9A-Za-z_.-]+)?\s+' \
+                             r'(?P<user>[0-9A-Za-z_.-]+)\s+' \
+                             r'(?:\d{4}-\d{2}-\d{2})\s+' \
+                             r'(?:\d{2}:\d{2}:\d{2})\s+' \
+                             r'(?P<job_state>[RWF])\s+' \
+                             r'(?P<queue>default|besteffort)'
+
+    def make_stat(self, orig_file, out_file, write_method):
+        with open(orig_file, 'r') as fin:
+            _ = fin.readline()  # header
+            fin.readline()  # dashes
+            re_match_positions = ('job_id', 'user', 'job_state', 'queue')
+            re_search = self.user_q_search
+            for line in fin:
+                qstat_values = self.process_line(re_search, line, re_match_positions)
+                self.l.append(qstat_values)
+        self.dump_all(out_file, self.stat_mapping[write_method])
+
+
+# qstat_mapping = {'yaml': (yaml.dump_all, {'Dumper': Dumper, 'default_flow_style': False}, 'yaml'),
+#                  'txtyaml': (qstat_write_lines, {}, 'yaml'),
+#                  'json': (json.dump, {}, 'json')}
+#
+# qstatq_mapping = {'yaml': (yaml.dump_all, {'Dumper': Dumper, 'default_flow_style': False}, 'yaml'),
+#                   'txtyaml': (qstatq_write_lines, {}, 'yaml'),
+#                   'json': (json.dump, {}, 'json')}
+#
+# pbsnodes_mapping = {'yaml': (yaml.dump_all, {'Dumper': Dumper, 'default_flow_style': False}, 'yaml'),
+#                     'txtyaml': (pbsnodes_write_lines, {}, 'yaml'),
+#                     'json': (json.dump, {}, 'json')}
```

Comment:
oops, maybe not; are these useful in other situations? 
if so, they should really not be commented out, just conditionally applied (or, selected via a dict structure :-D ).